### PR TITLE
allow apps to add an OCP GroupInterface backend

### PR DESCRIPTION
**Pull request by butonic**
~~less mess revealing~~ different solution than in https://github.com/owncloud/core/pull/21716

### CODE REVIEW

**Comment by icewind1991** [comment]
Looks good :+1:

File: lib/private/group/manager.php [hunk 1]
```diff
@@ -121,7 +122,7 @@ public function isBackendUsed($backendClass) {
 	}
 
 	/**
-	 * @param \OC_Group_Backend $backend
+	 * @param \OCP\GroupInterface $backend
 	 */
 	public function addBackend($backend) {
```
Comment (nickvergessen) [inline lib/private/group/manager.php]: Should we typehint the input?

Comment (butonic) [inline lib/private/group/manager.php]: hm makes sense but then we could typehint stuff all over the place. the user and group apis need an overhaul anyway, so i dont want to invest too much into this

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**